File: apps/nmp-gallery/android/app/src/main/kotlin/org/nmp/gallery/registry/KnownSigners.kt
package org.nmp.gallery.registry

/**
 * Ordered list of signers the Android detector knows about (detection
 * precedence = list order). Consumed by `detectInstalledSigners` in
 * ExternalSignerWire.kt (same package). Every `intentScheme` here MUST also
 * appear in `<queries>` in AndroidManifest.xml.
 */
val KNOWN_NOSTR_SIGNERS: List<NostrSignerInfo> = listOf(
    NostrSignerInfo(
        displayName = "Amber",
        intentScheme = "nostrsigner",
        contentAuthority = "com.greenart7c3.nostrsigner",
        packageName = "com.greenart7c3.nostrsigner",
    ),
    NostrSignerInfo(
        displayName = "Primal",
        intentScheme = "primal",
        contentAuthority = null,
        packageName = "net.primal.android",
    ),
)

File: crates/nmp-component-registry/registry/swiftui/login-block/KnownSigners.swift
extension NostrSignerDetector {

    /// Ordered list of signers this detector knows about (detection precedence
    /// = array order). Every `urlScheme` here MUST also appear in Info.plist's
    /// `LSApplicationQueriesSchemes`.
    @MainActor
    public static let knownSigners: [NostrSignerInfo] = [
        NostrSignerInfo(kind: .amber, displayName: "Amber"),
        NostrSignerInfo(kind: .primal, displayName: "Primal"),
        NostrSignerInfo(
            kind: .generic(name: "Nostr Connect", scheme: "nostrconnect"),
            displayName: "Nostr Connect"
        ),
    ]
}
